bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/SearchPattern.java

```diff
@@ -85,10 +94,33 @@ public class SearchPattern {
 	 * <code>"NPE"</code> string pattern, search will only use Camel Case match
 	 * rule, but with <code>N*P*E*</code> string pattern, it will use only Pattern
 	 * match rule.
-	 *
+	 * <p>
+	 * Unless {@link #RULE_SUBSTRING_MATCH} is in effect, it is required that
+	 * the 1st pattern's char must match the very 1st char of the search result.
```

Comment:
This doesn't need to be spec'd; this is the behavior one would expect according to other existing documentation.

Reply:
Are you sure about this? I couldn't find anything like that in the public documentation of this class. There is an example saying `NPE` will match `NullPointerException`, but it doesn't say it won't match for example `InvalidNullPointerException`.

Generally I think it really should be documented how concretely the new flag `RULE_SUBSTRING_MATCH` affects this and other behaviors (as `RULE_PATTERN_MATCH` above), somewhere - maybe moving this just to the new flag itself would be better? You say stuff like this shouldn't be part of the API contract - do I get it right that it means that we want to keep the flags "general" and kind of "not to propagate" how concretely the SearchPattern class behaves while doing concrete searches? Of course, if someone wants, they can find that info in the unit tests, but I'm still not sure if that suffices...

Reply:
ok